File: backend/booking-service/src/main/java/com/skylink/booking/service/BookingService.java
package com.skylink.booking.service;

import com.skylink.booking.dto.request.CreateBookingRequest;
import com.skylink.booking.dto.response.ApiResponse;
import com.skylink.booking.dto.response.BookingResponse;
import com.skylink.booking.kafka.event.PaymentEvent;

import java.util.List;

public interface BookingService {

    ApiResponse<BookingResponse> createBooking(CreateBookingRequest request);

    ApiResponse<List<BookingResponse>> getBookingsByUser(Long userId);

    ApiResponse<BookingResponse> getBookingById(Long bookingId);

    ApiResponse<String> cancelBooking(Long bookingId);

    void confirmBooking(PaymentEvent event);

    void failBooking(PaymentEvent event);

}

File: frontend/src/api/bookingService.ts
import axiosClient from './axiosClient';

export const bookingService = {
  createBooking: async (bookingData) => {
    const response = await axiosClient.post('/bookings', bookingData);
    return response.data;
  },
  getBookingsByUser: async (userId) => {
    const response = await axiosClient.get(`/bookings/user/${userId}`);
    return response.data;
  },
  getBookingById: async (id) => {
    const response = await axiosClient.get(`/bookings/${id}`);
    return response.data;
  },
  cancelBooking: async (id) => {
    const response = await axiosClient.patch(`/bookings/${id}/cancel`);
    return response.data;
  }
};
